### PR TITLE
chore: Add environment to gh workflows

### PR DESCRIPTION
**Pull request by kcadette**
Issue #, if available:

Description of changes:
Add environments to workflows so that we can require reviewers before actions are run.
To review a workflow from a PR, scroll down to checks and on each check click on details -> Review pending deployments -> Approve and deploy

The owner of the PR is only allowed to approve workflows AFTER the code review has been approved by a member of the team. Otherwise, only other team members can approve a workflow.

Checklist:

- [ ] Have you successfully deployed to an AWS account with your changes?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Did you add new integration tests?
- [ ] Did you run integration tests with your changes?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: common/config/rush/pnpm-config.json [hunk 1]
```diff
@@ -91,7 +91,8 @@
   "globalOverrides": {
     "jsonwebtoken@<=8.5.1": ">=9.0.0",
     "jsonwebtoken@<9.0.0": ">=9.0.0",
-    "xml2js@<0.5.0": ">=0.5.0"
+    "xml2js@<0.5.0": ">=0.5.0",
+    "yaml@<2.2.2": ">=2.2.2"
```
Comment (nguyen102) [inline common/config/rush/pnpm-config.json:95]: Is this change needed?

Comment (kcadette) [reply common/config/rush/pnpm-config.json]: I had the base branch: main
That change is in develop as a fix from pnpm audit

File: .github/workflows/smart-integ-inferno-test-periodically.yaml [hunk 2]
```diff
@@ -3,6 +3,10 @@ on:
   schedule:
     - cron: '0 13 1,15 * *'
 
+env:
```
Comment (ssvegaraju) [inline .github/workflows/smart-integ-inferno-test-periodically.yaml:6]: you may need to pull changes to revert these: we pivoted away from using AWS_ACCESS_KEY_ID in the secrets and instead use the github action for sts short-term credentials.

**Review by ssvegaraju — COMMENTED** [review]
LGTM: just one file that needs to be updated

File: .github/workflows/smart-integ-inferno-test-periodically.yaml [hunk 3]
```diff
@@ -127,21 +114,3 @@ jobs:
         run: |
           cd ./solutions/smart-deployment
           node ../../common/scripts/install-run-rushx.js int-test
-  slack-notification:
```
Comment (Bingjiling) [inline .github/workflows/smart-integ-inferno-test-periodically.yaml:130]: Why is this removed?

Comment (kcadette) [reply .github/workflows/smart-integ-inferno-test-periodically.yaml]: I needed to pull changes from develop

File: .github/workflows/smart-integ-inferno-test-periodically.yaml [hunk 4]
```diff
@@ -67,7 +68,7 @@ jobs:
   custom-integration-tests:
     needs: inferno-test
     name: Run custom integration tests - enableMultiTenancy=${{ matrix.enableMultiTenancy }}
-    environment: FWoA Integ Test Env
```
Comment (Bingjiling) [inline .github/workflows/smart-integ-inferno-test-periodically.yaml:70]: Would this affect the execution of the workflow? Does the workflow depends on FWoA Integ Test Env for anything?

**Review by Bingjiling — APPROVED** [review]
